include/s3select_functions.h

```diff
@@ -782,6 +782,18 @@ struct _fn_extact_from_timestamp : public base_function
     {
       result->set_value( (int64_t)new_ptime.date().week_number() );
     }
+    else if( strcmp(val_date_part.str(), "hour")==0 )
```

Comment:
please review high level design doc. 
timestamp functions interface should be modified, not to include string expression "hour" and such.
this syntax should be rejected by engine.

Reply:
expressions should be spaced normally "foo == 0"